php/libraries/NDB_BVL_Battery.class.inc

```diff
@@ -312,30 +312,18 @@ class NDB_BVL_Battery
         if (!is_null($stage)) {
             $query .= ", b.instr_order as instrument_order";
         }
-        $query  .= " FROM flag AS f, test_names as t, test_subgroups as ts ";
+        $query  .= " FROM flag f
+            	JOIN test_names t ON (f.Test_name=t.Test_name)
+            	JOIN test_subgroups ts ON (ts.ID = t.Sub_group) ";
         $qparams = array('SID' => $this->sessionID);
 
         if (!is_null($stage)) {
-            $query .= ", test_battery AS b";
+            $query .= "
+            	LEFT JOIN test_battery b ON (t.Test_name=b.Test_name) ";
         }
-        $query .= " WHERE ts.ID = t.Sub_group
-            AND f.Test_name=t.Test_name
-            AND f.SessionID=:SID
+        $query .= " WHERE f.SessionID=:SID
             AND LEFT(f.CommentID, 3) != 'DDE'";
 
-        if (!is_null($stage)) {
-            $query .= " AND t.Test_name=b.Test_name";
-
-            if (!is_null($SubprojectID)) {
-                $query .= " AND (
-                    b.SubprojectID IS NULL
-                    OR b.SubprojectID=:SubprojID
-                )";
```

Comment:
I don't think this if statement should be removed. It completely changes the behaviour and will break scripts like assign missing instruments.php